src/libraries/Microsoft.PowerFx.Interpreter/Marshal/TypeCoercionProvider.cs

```diff
@@ -238,16 +238,16 @@ internal static bool TryCoerceTo(this FormulaValue value, FormattingInfo formatI
         public static bool TryCoerceToRecord(this RecordValue value, RecordType targetType, out RecordValue result)
         {
             result = null;
-            int n = value.Fields.Count();
-            var recordResult = new NamedValue[n];
-
-            for (int i = 0; i < n; i++)
+            var recordResult = new List<NamedValue>();
+            
+            foreach (var sourceField in value.Fields)
             {
-                var fieldName = value.Fields.ElementAt(i).Name;
+                var fieldName = sourceField.Name;
                 var fieldValue = value.GetField(fieldName);
                 if (!targetType.TryGetFieldType(fieldName, out FormulaType fieldType))
                 {
-                    return false;
+                    recordResult.Add(new NamedValue(fieldName, new BlankValue(IRContext.NotInSource(FormulaType.Blank))));

```

Comment:
Now we will return result with a union of value type and targetType, is it expected?

I feel it could lead to unexpected scenario. Imagine this: we have A `{ key : value, keya: value }` and B `{ key: value, keyb: value}` and we want to coerce them both to C `{ key: value}`. But now we have new A and new B of different types.